django/middleware/common.py

```diff
@@ -38,10 +38,13 @@ def process_request(self, request):
         """
 
         # Check for denied User-Agents
-        if 'HTTP_USER_AGENT' in request.META:
+        try:
+            uagent = request.META['HTTP_USER_AGENT']
             for user_agent_regex in settings.DISALLOWED_USER_AGENTS:
-                if user_agent_regex.search(request.META['HTTP_USER_AGENT']):
+                if user_agent_regex.search(uagent):
                     raise PermissionDenied('Forbidden user agent')
+        except KeyError:
+            pass
```

Comment:
We usually use `get` to avoid the try boilerplate

```python
user_agent = request.META.get('HTTP_USER_AGENT')
if user_agent is not None:
    for user_agent_regex in settings.DISALLOWED_USER_AGENTS:
        if user_agent_regex.search(user_agent):
          raise PermissionDenied('Forbidden user agent')

Reply:
@charettes , yeah, that definitely crossed my mind.  Will update.

Reply:
Redid in https://github.com/django/django/pull/11144 so as not to clutter up with an unnecessary commit.